cookbooks/Getting-Started/eval/promptfoo/README.md

```diff
@@ -0,0 +1,29 @@
+# Promptfoo integration
+
+Use case: I'm a SWE who wants to run my AIConfig against a set of test cases specified in a config file. Each test case has the input and a success condition of my choosing.
+
+## Philosophy / design
+
+Prompfoo has a pretty nice interface (both input and outputs) for addressing the use case. Tests are specified in a yaml file and the test suite can be run with a simple command. The same config file makes it easy to connect your test suite to an AI config with a small amount of code.
+
+## How-to guide
+
+1. Make a file `prompt.txt` that just has the contents
+
+   `"{{question}}"`
+   
+    (literally, don't put any actual question there).
+2. Write your test cases in a Promptfoo config file. See promtfooconfig.yaml as an example. Don't worry about "providers" for now; we'll get to that.
+3. Write a "connector" script to tell Promptfoo how to test your AIConfig. For example, `travel.aiconfig.json` has the corresponding `travel.py` script.
```

Comment:
This is where I think we can provide a simple interface that takes an aiconfig (either file or object), prompt to run, parameter overrides, and whether to run with dependencies or not (I.e. same signature as run function).

Then people don't need to create this script themselves for the vast majority of use cases.

Reply:
OK, let me take a swing at this.

Reply:
> parameter overrides, and whether to run with dependencies or not (I.e. same signature as run function).


I'm a little confused about this part - AIConfigRuntime.run takes the prompt name, inputs (aka params), and inference options that I don't think are relevant for testing, right? What's the run with dependencies option? cc @Ankush-lastmile

Reply:
run_with_dependencies is a kwarg you pass into run() to invoke run_with_dependencies.

I think the idea here is that the interface is built on top of config.run(), and all the attributes for run are passed through from the interface

Reply:
Right, I see. I think for testing we need run_with_dependencies==False exactly when we want to unit test a config. i.e. pass a test suite specific inputs to one prompt without running its dependencies. 

Let's note that this is a special case of running an arbitrary sub-DAG, which we probably want to support eventually.  So we will want to generalize run_with_dependencies to something like run_with_leaves() where leaves sets the input params for the sub-DAG we want to run.

Reply:
We have an indirection layer now (settings.json) where we can add arbitrary config as needed like run_with_dependencies=False. Can that be a follow up?